janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -5827,6 +5865,39 @@ public void testVertexCentricQueryProfiling() {
         }};
         mSingleLabel.getAnnotations().remove("percentDur");
         assertEquals(annotations, mSingleLabel.getAnnotations());
+
+        // satisfied by a single vertex-centric query which is satisfied by union of two edge queries
+        newTx();
+        Metrics mMultiLabels = tx.traversal().V(v).outE("friend", "friend-no-index").has("time", 100)
+            .profile().next().getMetrics(1);
+        assertEquals("JanusGraphVertexStep([time.eq(100)])", mMultiLabels.getName());
+        assertTrue(mMultiLabels.getDuration(TimeUnit.MICROSECONDS) > 0);
+        annotations = new HashMap() {{
+            put("condition", "(time = 100 AND (type[friend] OR type[friend-no-index]))");
+            put("orders", "[]");
+            put("vertices", 1);
+        }};
+        mMultiLabels.getAnnotations().remove("percentDur");
+        assertEquals(annotations, mMultiLabels.getAnnotations());
+        assertEquals(3, mMultiLabels.getNested().size());
+        Metrics friendMetrics = (Metrics) mMultiLabels.getNested().toArray()[1];
+        assertEquals("OR-query", friendMetrics.getName());
+        // FIXME: assertTrue(friendMetrics.getDuration(TimeUnit.MICROSECONDS) > 0);
+        annotations = new HashMap() {{
+            put("isFitted", "true");
+            put("isOrdered", "true");
+            put("query", "2069:byTime:SliceQuery[0xB0E0FF7FFFFF9B,0xB0E0FF7FFFFF9C)@2147483647"); // vertex-centric index utilized
```

Comment:
Not specific to this line, but I put it here as an example:

Although I like how much effort you put into testing the profiler output, I expect we will have a hard time adapting all test cases if TinkerPop only ever so slightly changes the format of the profiler output :sweat_smile: 
That's only a quick thought that came to my mind, not saying that I have any idea how to do it better.

Reply:
I got your point. I totally understand sometimes a change detector is not a good test. However, I still did this because:
1) For this line specifically, we have to assert because otherwise there is no way to assert whether vertex-centric index is used or not. ("byTime" is the index name and "2069" is its encoded version. Btw I really don't want to include "2069" but I cannot do much about it unless I change how index meta-data is stored, which I don't want to touch at the moment due to risk concern). 
2) In general, there is no much Tinkerpop format specific thing here. All these string literals like "isFitted"/"isOrdered"/"query" are all JanusGraph annotations. The hex string "0xB0E0FF7FFFFF9B" is calculated by JanusGraph and it won't change unless we change the way JanusGraph stores data (which means this line effectively prevents people from accidentally changing data format, which might not be detected by other tests). Of course, if we literally test the output of profiling result in string format, as pasted in https://github.com/JanusGraph/janusgraph/pull/2410#issuecomment-770188509, it would be very sensitive to any small change to Tinkerpop format.
The major Tinkerpop-coupled thing is the structure, i.e. how things are grouped and nested. This is unavoidable and does make sense because JanusGraph profiling-relevant code is already coupled with Tinkerpop profiling-relevant components. 

Currently, there are some test cases checking profiling result, but those are very basic that only check, say, whether backend-query exists, whether a certain annotation exists. Thus I added these detailed tests. Let me know if you have another idea of how to test them.